dance/pipeline.py

```diff
@@ -856,7 +857,58 @@ def flatten_dict(d, *, parent_key="", sep="_"):
     for k, v in d.items():
         new_key = parent_key + sep + k if parent_key else k
         if isinstance(v, dict):
-            items.extend(flatten_dict(v, new_key, sep).items())
+            items.extend(flatten_dict(d=v, parent_key=new_key, sep=sep).items())
         else:
             items.append((new_key, v))
     return dict(items)
+
+
+def generate_combinations_with_required_elements(elements, required_indexes):
+    required_elements = [elements[required_index] for required_index in required_indexes]
+    optional_elements = [x for x in elements if x not in required_elements]
+
+    # Sort optional elements in the same order as in the `elements` list
+    optional_elements.sort(key=lambda x: elements.index(x))
+
+    # Generate all possible combinations of optional elements
+    optional_combinations = []
+    for i in range(len(optional_elements) + 1):
+        optional_combinations += list(itertools.combinations(optional_elements, i))
+
+    # Combine required elements with optional combinations to get all possible combinations
+    all_combinations = []
+    for optional_combination in optional_combinations:
+        all_combinations.append([x for x in elements if x in required_elements or x in optional_combination])
+    return all_combinations
+
+
+def generate_subsets(path, tune_mode, save_directory, required_indexes):
+    """Generated automatically subsets for original yaml. YAML can be generated
```

Comment:
This is very unclear. Perhaps you can give an example besides the verbal description. I can help improve the descriptions later.